### ISSUE TITLE
Return the mask tensor from function to_dense_batch

### ISSUE THREAD

**Issue by pietroastolfi** [rusty1s/pytorch_geometric#1800]
## 🚀 Feature

Return the mask tensor from function `torch_geometric.utils.to_dense_batch`, so that it's super fast to go back to an "unpadded" representation.

## Motivation

The function `torch_geometric.utils.to_dense_batch` is very useful, but in the api I did not find the opposite direction function. 

## Additional context

Right now my solution to go back to an unpadded representation is:

```
def unpad_tensor(x, lengths, dim=0):
    M = lengths.max()
    slices = []
    for i in lengths:
        slices += [i, M-i]
    unpad_x = torch.split(x, slices, dim=dim)[::2]
    
    return torch.cat(unpad_x, dim=dim)
```

**Comment by rusty1s** [comment]
Thanks for this issue. Note that `to_dense_batch` already returns a mask, and going back is quite straightforward:
```python
x_dense, mask = to_dense_batch(x, batch)
x = x_dense.view(-1, x.size(-1))[mask.view(-1)]
```
We could wrap this functionality into its own function. Feel free to create a PR if you like :)

**Comment by pietroastolfi** [comment]
OH sorry!! I never realized that the function was returning a mask. I've always been mislead by the doc that says it returns a tensor [N_0, ..., N_b]. 

At this point, I think that fixing the doc is enough instead of creating a specific function. Once you know that `to_dense_batch ` returns a mask, the use of it to retrieve the original representation is so immediate (as your 1 line code shows) that it does not deserve a specific function.
